datadog/resource_datadog_logs_metric.go

```diff
@@ -0,0 +1,324 @@
+	if err != nil {
+		return translateClientError(err, "error creating LogsMetric")
+	}
+	id := *response.GetData().Id
+	d.SetId(id)
+
+	return resourceDatadogLogsMetricRead(d, meta)
+}
+
+func resourceDatadogLogsMetricRead(d *schema.ResourceData, meta interface{}) error {
+	providerConf := meta.(*ProviderConfiguration)
+	datadogClient := providerConf.DatadogClientV2
+	auth := providerConf.AuthV2
+	var err error
+
+	id := d.Id()
+
+	resourceLogsMetricResponse, httpResp, err := datadogClient.LogsMetricsApi.GetLogsMetric(auth, id).Execute()
+
+	if err != nil {
+		if httpResp.StatusCode == 404 {
+			// this condition takes on the job of the deprecated Exists handlers
+			d.SetId("")
+			return nil
+		}
+		return translateClientError(err, "error reading LogsMetric")
+	}
+
+	resource := resourceLogsMetricResponse.GetData()
+
+	if ddAttributes, ok := resource.GetAttributesOk(); ok {
+		if computeDDModel, ok := ddAttributes.GetComputeOk(); ok {
+			computeMap := map[string]interface{}{}
+			if v, ok := computeDDModel.GetAggregationTypeOk(); ok {
+				computeMap["aggregation_type"] = *v
+			}
+			if v, ok := computeDDModel.GetPathOk(); ok {
+				computeMap["path"] = *v
+			}
+			d.Set("compute", []map[string]interface{}{computeMap})
+		}
+		if filterDDModel, ok := ddAttributes.GetFilterOk(); ok {
+			filterMap := map[string]interface{}{}
+			if v, ok := filterDDModel.GetQueryOk(); ok {
+				filterMap["query"] = *v
+			}
+			d.Set("filter", []map[string]interface{}{filterMap})
+		}
+		if groupByArray, ok := ddAttributes.GetGroupByOk(); ok {
+			mapAttributesArray := make([]map[string]interface{}, 0)
+
+			for _, groupByArrayItem := range *groupByArray {
+				mapAttributesArrayIntf := map[string]interface{}{}
+				if v, ok := groupByArrayItem.GetPathOk(); ok {
+					mapAttributesArrayIntf["path"] = *v
+				}
+				if v, ok := groupByArrayItem.GetTagNameOk(); ok {
+					mapAttributesArrayIntf["tag_name"] = *v
+				}
+
+				mapAttributesArray = append(mapAttributesArray, mapAttributesArrayIntf)
+			}
+			d.Set("group_by", mapAttributesArray)
+		}
+	}
+
+	if v, ok := resource.GetIdOk(); ok {
+		d.Set("name", *v)
+	}
+
+	return nil
+}
+
+func buildDatadogLogsMetricUpdate(d *schema.ResourceData) (*datadogV2.LogsMetricUpdateData, error) {
+	result := datadogV2.NewLogsMetricUpdateDataWithDefaults()
+	attributes := datadogV2.NewLogsMetricUpdateAttributesWithDefaults()
+
+	filter, err := getFilter(d)
+	if err != nil {
+		return nil, err
+	}
+	attributes.SetFilter(*filter)
+
+	groupBys, err := getGroupBys(d)
+	if err != nil {
+		return nil, err
+	}
+	attributes.SetGroupBy(groupBys)
+
+	result.SetAttributes(*attributes)
+
+	return result, nil
+}
+
+func resourceDatadogLogsMetricUpdate(d *schema.ResourceData, meta interface{}) error {
+	providerConf := meta.(*ProviderConfiguration)
+	datadogClient := providerConf.DatadogClientV2
+	auth := providerConf.AuthV2
+
+	resultLogsMetricUpdateData, err := buildDatadogLogsMetricUpdate(d)
```

Comment:
Let's do something with the error here, like above in the  create method

Reply:
Indeed, thx 👍